### PR TITLE
Update to .vue style

### PR DESCRIPTION
**Pull request by pangaunn**

### CODE REVIEW

**Comment by ches** [comment]
Hi @pangaunn, thank you for contributing! I'll leave the feedback to @waf as this project is foremost his baby so far, but could you provide a description for the PR please? It's always nice for a pull request to come with some explanation of what you've changed, and why you feel the change is the right thing to do.

**Comment by pangaunn** [comment]
Hi @ches, I'm changing one html file into multiple vue component file. 95% of that is @waf's code. I just rearrange it.

**Comment by waf** [comment]
@pangaunn thanks for adding this! This looks great. I unfortunately cannot attend bkkhack tonight. I'll review this fully soon, though!

From a very quick review, here are some questions, mostly involving making it easy to collaborate with other people:

1. The package.json has changed all the commands, but the readme has not been updated
   - How do we build a release build (minified) vs a non-release build?
   - How do we ensure that the `gh-pages` branch always has the release build? The old code had a `git name-rev` command that ensured this.
2. It looks like all the javascript is now in .vue files. How do common editors get syntax highlighting for javascript, html, css? Is there some editor setup page you can add a hyperlink to?
2. It looks the initial page is an empty div, so the user needs to wait for the javascript to run before they can see anything. Is that correct? Can we change this so we at least render the 2 sidebars, center column, title, etc without javascript? It's fine to wait for dynamic content, but it's weird to wait for static content.
4. Was changing all the indentation from 4-spaces to 2-spaces intentional? If it was a mistake, we should stick with the original formatting. If you really want to make this change, then that's fine.

**Comment by pangaunn** [comment]
1. I will update readme soon. 
  1.1 For release build you can run `npm  run build`. For development you can run `npm start` or `npm run dev`
  1.2 I will take a look with old code command and will update this pull request.
2. I'm using vetur extension in visual code. Most of editor has .vue file syntax highlight.
3. This is how SPA works. If you not ok with this I will try some kind of loading indicator.
4. Yes that from standard eslint.

**Comment by waf** [comment]
Great, thanks so much for replying so quickly! Can you please add items 1 and 2 to the readme?

For item 3, let's not add a loading indicator. I think we can maintain SPA behavior, but still [optimize for first paint](https://developers.google.com/web/tools/lighthouse/audits/first-meaningful-paint), which is an important way to make SPAs not seem slow. I'll take a look at this and push to this PR.

I'll also do some benchmarking for load time between the current `master` branch and this PR. It doesn't have to be the exact same speed, but hopefully it won't be e.g. 2x slower.

**Comment by waf** [comment]
@pangaunn I've had some more time to go through the code, and I'm really impressed by the code quality and decoupling of components. Nice work!

**Comment by pangaunn** [comment]
Please check @waf I already commit changes.

File: src/index.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ model.projects.loading = true;
 
 // encapsulates access to the github issues API
 const githubIssue = new GitHubIssueService({
-    organization: "bkkhack",
+    organization: "pangaunn",
```
Comment (ches) [inline src/index.js]: Let's not forget to change this back before merge 😄

**Review by ches — COMMENTED** [review]
Seems like a lot of positive improvement, but I would echo the sentiment that it would be courteous to check with project maintainers about opinionated matters like code formatting—indentation, quoting style, semicolons, etc.—before changing them unilaterally in a pull request. Same goes for substantial changes to build automation, maintainers are often going to expect these to be reasoned and discussed separately from implementation changes. Just my advice for your adventures in open source 😁 

Also, I'm assuming we probably want the `dist` directory gitignored outside of the `gh-pages` branch, it contains generated artifacts and they're very noisy to source control history and reviews.

**Comment by ches** [comment]
Separately, @waf this brings to mind that it'd be a good time for you to apply a license to the project ☺️

**Comment by pangaunn** [comment]
@ches Thanks for the advice. For a coding style, I didn't think before sorry @waf.

**Comment by waf** [comment]
@ches thanks. opened up https://github.com/bkkhack/hackmap/pull/11 to discuss the licensing /cc @djay 

@pangaunn no problem, thanks!

**Comment by pangaunn** [comment]
Please check I've update .gitignore to show build folder and ignore dist folder.
For src/index.js, It didn't use anymore.

**Comment by waf** [comment]
Thanks @pangaunn. I see how the run this using the development server, but how do we run this in production mode, locally? 

I did the following, from the readme:

```bash
> npm install
> npm run build
```

and got a notification that the minified sources are built to the `dist` folder. I didn't see a way to run that using npm, so I used `python3 -m http.server` from the `dist` directory. However, then the filepaths aren't correct for the css and js, so the app fails to load:

image

Thanks!

**Comment by pangaunn** [comment]
`npm run build` will put /hackmap prefix when build.
I've added another command `npm run build:local` to build without put /hackmap prefix.

**Comment by pangaunn** [comment]
@waf, Any update on this PR? Feel free to tell me what I need to update.  : )

**Comment by waf** [comment]
Hi @pangaunn, sorry, I was traveling this past weekend. I will get to this :)

**Comment by waf** [comment]
I've reviewed this PR, and it looks very good. Thank you for converting this, and decoupling the components. The code quality is good. Just some minor questions, mostly related to maintenance.

- `/hackmap` prefix - is there a way to make it not need hackmap? If everything's relative urls, we shouldn't need special cases like `build` vs `build:local`
- There's about 600 or 700 lines of new javascript for build scripts. Did you hand-write this yourself or are they generated? How do we keep these scripts up to date?

Thanks!

**Comment by pangaunn** [comment]
1. I add /hackmap because production url has /hackmap https://bkkhack.github.io/hackmap/
2. They generated.

**Comment by waf** [comment]
1. That's a Github Pages url. There's two different git repositories, one at `/` and one (this repo) at `/hackmap/`
    - https://github.com/bkkhack/bkkhack.github.io is deployed to https://bkkhack.github.io/ as the root of the website.
    - https://github.com/bkkhack/hackmap/ is deployed to https://bkkhack.github.io/hackmap/ so the root of this application will be deployed to `/hackmap/` (not `/`) so we can just not specify any absolute urls and it will work.
2. From where? How do we update them?

**Comment by pangaunn** [comment]
1. I will try to fix it. Normally when build for production index.html will required file like /static/js so it can't find on github server. All I need to do is just update it to require static/js right?
2. It was generate from `vue init  webpack`

**Comment by pangaunn** [comment]
I've fixed /hackmap prefix please check. For build script, I think it rarely to see it update. If it update I will update it here too.

**Comment by ches** [comment]
This page cleared up for me a lot of what I guess are norms in the Vue ecosystem that we see here in this PR: <https://vuejs.org/v2/guide/single-file-components.html>

And all the generated Webpack build boilerplate stems from <https://github.com/vuejs-templates/webpack>.

**Comment by waf** [comment]
Thanks @pangaunn! There were some merge conflicts due to recent changes to hackmap. I'm merging your PR to the `update-to-vue-style` branch, where I will fix the merge conflicts, then merge that to master. Thank you for updating this and teaching me best practices for vue!

**Comment by pangaunn** [comment]
You're welcome, I'm glad to help the community.  : )

**Pull request merged**